client/question/components/LearnerViewDirective.js

```diff
@@ -795,20 +795,22 @@ tie.directive('learnerView', [function() {
 
         $scope.onVisibilityChange = function() {
           var question = CurrentQuestionService.getCurrentQuestion();
-          var tasks = question.getTasks();
-
-          // When a user changes tabs (or comes back), add a SessionPause
-          // or SessionResumeEvent, respectively.
-          var hiddenAttributeName = (
-            $scope.determineHiddenAttributeNameForBrowser());
-          if (hiddenAttributeName !== null && tasks !== null &&
-              tasks.length > currentTaskIndex) {
-            if ($scope.isDocumentHidden(hiddenAttributeName)) {
-              EventHandlerService.createSessionPauseEvent(
-                tasks[currentTaskIndex].getId());
-            } else {
-              EventHandlerService.createSessionResumeEvent(
-                tasks[currentTaskIndex].getId());
+          if (question) {
```

Comment:
What's the plan in the "else" case?

Reply:
Nothing should happen? That means the question isn't initialized yet and trying to put a task ID for an uninitialized question is undefined behavior.

I could set the task id to be "pre-question", if you want, but I'm not sure that would be captured properly in the analytics. Whatcha think?

Reply:
This fix is mostly for testing -- the pages manage to switch before the question is fully loaded (or it doesn't load, in some cases), which cause getTasks() to be called on a null.